flax/linen/module.py

```diff
@@ -1474,7 +1477,10 @@ def clone_fn(m: Module) -> Module:
           if key in cache:
             return cache[key]
           else:
-            clone = m.clone(_deep_clone=cache)
+            if _update_children:
+              clone = m.clone(_deep_clone=cache, _update_children=_update_children, **updates)
```

Comment:
Not entirely sure about this. We are passing the `updates` for the top-level Module to its submodules, there is a high chance they receive different arguments.